musmart/timing/tempo_modeling.py

```
@@ -0,0 +1,146 @@
+
+    The tempo drift is the gradient of the slope of signed overall tempo change in a performance,
+    such that larger values imply a greater departure from a linear tempo slope (i.e., not just
+    accelerating or decelerating).
+
+    Parameters
+    ----------
+    beats : Iterable
+        An iterable of beat timestamps in seconds corresponding to, e.g., quarter notes.
+
+    Returns
+    -------
+    float
+        The tempo drift value.
+
+    Raises
+    ------
+    ValueError
+        If linear regression cannot be calculated.
+
+    Examples
+    --------
+    >>> stable = [1., 2., 3., 4.]
+    >>> unstable = [1., 2.1, 2.9, 3.5]
+    >>> tempo_drift(stable) < tempo_drift(unstable)
+    True
+
+    """
+
+    return float(_fit_tempo_linregress(beats).stderr)
+
+
+def tempo_fluctuation(beats: Iterable) -> float:
+    """
+    Calculates the percentage fluctuation about the overall tempo of provided `beats`.
+
+    Tempo fluctuation can be calculated as the standard deviation of the tempo of a
+    performance normalized by the mean tempo [1].
+
+    Parameters
+    ----------
+    beats : iterable
+        An iterable of beat timestamps in seconds corresponding to, e.g., quarter notes.
+
+    Returns
+    -------
+    float
+        The tempo fluctuation value.
+
+    Examples
+    --------
+    >>> my_beats = [1., 2., 3., 4.]  # stable performance
+    >>> tempo_fluctuation(my_beats)
+    0.0
+
+    References
+    ----------
+    [1] Cheston, H., Schlichting, J.L., Cross, I., and Harrison, P.M.C. (2024).
+        Jazz Trio Database: Automated Annotation of Jazz Piano Trio Recordings Processed Using
+        Audio Source Separation. Transactions of the International Society for Music Information
+        Retrieval, 7/1 (pp. 144–158). https://doi.org/10.5334/tismir.186
+
+    """
+
+    inter_beat_intervals = np.diff(beats)
+    return float(np.nanstd(inter_beat_intervals) / np.nanmean(inter_beat_intervals))
```

Why are you silently dropping NaNs? Why would you have NaNs here?

---

Fixed in https://github.com/music-computing/amads/pull/85, we explicitly raise an error on NaN values